envisage/extension_point_binding.py

```diff
@@ -23,15 +23,15 @@ class ExtensionPointBinding(HasTraits):
     #### 'ExtensionPointBinding' interface ####################################
 
     # The object that we are binding the extension point to.
-    obj = Any
+    obj = WeakRef
```

Comment:
Similarly, all uses of `self.extension_registry` need to be examined, and possibly guarded with a `None` check.